### DISCUSSION TITLE
LangSmith tracing for basic Prompt/Question & OpenAI LLM app & On-Prem or local hosting

### DISCUSSION THREAD

**Discussion by mansats** [langchain-ai/langsmith-sdk#504] [Ideas]
I am freelancer, wanted to build a very basic question/prompt and response from OpenAI llm. Appreciate help on the following questions.
1) Can I still use LangSmith's tracing for this simple app? 
2) Is it feasible to host entire tracking (storing & monitoring of trace) on my environment instead of LangSmith's cloud? 
3) if yes is the answer for both the questions, what are the tech stack required?

**Comment by hinthornw** [comment]
1. Yes langsmith tracing works great with this.
2. Self-hosting currently requires an enterprise license, which doesn't seem appropriate here (at first glance)

I'd recommend using the hosted version of LangSmith here if you want this functionality.
